### PR TITLE
Fixes Shuttle Crushed Huggers

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Puts an end to the hugger trap shuttle crush bug for good.

## Why It's Good For The Game

As above.

Fixes: https://github.com/tgstation/TerraGov-Marine-Corps/issues/5660

## Changelog
:cl:
fix: Hugger traps crushed by the shuttle no longer discharge huggers (again).
/:cl:

### CODE REVIEW

File: code/game/objects/structures/xeno.dm [hunk 1]
```diff
@@ -131,9 +131,17 @@
...
+		drop_hugger()
+
+	return ..()
+
+///No more trapping shuttles with huggies
+/obj/effect/alien/resin/trap/proc/shuttle_crush()
 	SIGNAL_HANDLER
-	QDEL_NULL(hugger)
+	qdel(hugger)
+	hugger = null
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:144]: QDEL_NULL

**Review by TiviPlus — CHANGES_REQUESTED** [review]
In future avoid strong cursing please I dont wanna risk getting swatted by github

File: code/game/objects/structures/xeno.dm [hunk 2]
```diff
@@ -131,9 +131,17 @@
...
+	if(hugger && loc)
+		drop_hugger()
+
+	return ..()
+
+///No more trapping shuttles with huggies
+/obj/effect/alien/resin/trap/proc/shuttle_crush()
 	SIGNAL_HANDLER
 	QDEL_NULL(hugger)
+	hugger = null
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:144]: Unneeded

Comment (Surrealaser) [reply code/game/objects/structures/xeno.dm:144]: This is literally what made the difference so it is very much needed.

File: code/game/objects/structures/xeno.dm [hunk 3]
```diff
@@ -131,9 +131,17 @@
...
 
-/obj/effect/alien/resin/trap/proc/shuttle_crush() //No more trapping shuttles with huggies
+/obj/effect/alien/resin/trap/obj_destruction(damage_flag)
+	if(hugger && loc)
+		drop_hugger()
+
+	return ..()
+
+///No more trapping shuttles with huggies
+/obj/effect/alien/resin/trap/proc/shuttle_crush()
```
[the remark this answers is not in the document]
Comment (Surrealaser) [reply code/game/objects/structures/xeno.dm:141]: Didn't you want this for documentation initially?

File: code/game/objects/structures/xeno.dm (hunk 2 above)
Comment (TiviPlus) [reply code/game/objects/structures/xeno.dm:144]: Qdel null Deletes and then nulls a variable I reccomend you go read what it does

File: code/game/objects/structures/xeno.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/game/objects/structures/xeno.dm:141]: What?
Documentation explains what the proc does not "No more trapping shuttles with huggies"
An example of proper documentation is
"Called when the trap is crushed by a shuttle, deletes the hugger inside"

File: code/game/objects/structures/xeno.dm [hunk 4]
```diff
@@ -775,14 +777,19 @@ TUNNEL
 	creator = null
 	return ..()
 
+///Ensures that no acid gas will be released when the well is crushed by a shuttle
+/obj/effect/alien/resin/acidwell/proc/shuttle_crush()
+	SIGNAL_HANDLER
+	QDEL_NULL(src)
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:783]: Just qdel no need to null

File: code/game/objects/structures/xeno.dm [hunk 5]
```diff
@@ -131,9 +131,17 @@
...
+	if(damage_amount && hugger && loc)
+		drop_hugger()
+
+	return ..()
+
+///Ensures that no huggies will be released when the trap is crushed by a shuttle; no more trapping shuttles with huggies
+/obj/effect/alien/resin/trap/proc/shuttle_crush()
 	SIGNAL_HANDLER
-	QDEL_NULL(hugger)
+	QDEL_NULL(src)
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:143]: Ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/xeno.dm (hunk 4 above)
Comment (Surrealaser) [reply code/game/objects/structures/xeno.dm:783]: I don't trust that assertion; I'm testing this first.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**